File: shinyan/boj/20546.py
import sys

n = int(sys.stdin.readline()) 
stockprice = list(map(int,sys.stdin.readline().split()))

def J():
    count = 0
    nowcash = n
    global jun_assets

    for stock in stockprice:
        count += nowcash // stock #남은 돈
        nowcash = nowcash % stock #매수가능 주식 수
            
        if nowcash == 0:
            break

    jun_assets = nowcash + count * stockprice[-1]

    return nowcash, count

def S():
    count_stock = 0
    nowcash = n
    global sung_assets

    for i in range(len(stockprice)-4):
        if(stockprice[i] < stockprice[i+1] < stockprice[i+2] < stockprice[i+3]):
            nowcash += count_stock * stockprice[i+3]
        
        elif(stockprice[i] > stockprice[i+1] > stockprice[i+2] > stockprice[i+3]):
            count_stock += nowcash // stockprice[i+3]
            nowcash = count_stock % stockprice[i+3]
        
    sung_assets = nowcash + count_stock * stockprice[-1]

    return count_stock, nowcash

J()
S()

if jun_assets > sung_assets:
    print("BNP")

elif jun_assets < sung_assets:
    print("TIMING")

else:
    print("SAMESAME")

File: kangrok/boj-20546/20546.swift
let money = Int(readLine()!)!

let chart = readLine()!.split(separator: " ").map { Int($0)! }

var J: (money: Int, stock: Int) = (money, 0)
var S: (money: Int, stock: Int) = (money, 0)
var downCount = 0
var upCount = 0

for i in 0..<chart.count {
  if i != 0 {
    if chart[i] > chart[i-1] {
      upCount += 1
      downCount = 0
    } else if chart[i] == chart[i-1] {
      upCount = 0
      downCount = 0
    } else {
      downCount += 1
      upCount = 0
    }
  }
  
  if J.money >= chart[i] {
    J.stock += J.money / chart[i]
    J.money -= J.money / chart[i] * chart[i]
  }
  
  if downCount >= 3 && S.money >= chart[i] {
    S.stock += S.money / chart[i]
    S.money -= S.money / chart[i] * chart[i]
  }
  
  if upCount >= 3 && S.stock > 0 {
    S.money += S.stock * chart[i]
    S.stock = 0
  }
  
  if i == 13 {
    if J.money + J.stock * chart[i] > S.money + S.stock * chart[i] {
      print("BNP")
    } else if J.money + J.stock * chart[i] < S.money + S.stock * chart[i] {
      print("TIMING")
    } else {
      print("SAMESAME")
    }
  }
}
